src/tamp_improv/benchmarks/blocks2d_env.py

```diff
@@ -1,6 +1,6 @@
 """A block environment in 2D."""
 
-from typing import Any, SupportsFloat, Tuple
+from typing import Any, Dict, Optional, Tuple
```

Comment:
instead of importing `Dict` and `Tuple`, you can just use `dict` and `tuple` for typing annotations (as of more recent versions of python -- I think 3.8+). The same goes for `list`.

Also, instead of importing `Optional`, you can do the following: instead of `Optional[int]`, you can write `int | None`, which means "int or None" (the `|` is a way to specify union types).